src/modalities/config/component_factory.py

```diff
@@ -0,0 +1,129 @@
+from typing import Any, Dict, List, Union
+
+from pydantic import RootModel
+
+from tests.config.hierarchical_instantiation.configs import CompConfigABC
```

Comment:
This should be `from tests.config.configs import CompConfigABC`. However, maybe we should consider moving the `CompConfigABC` class from `tests` to `modalities`, as it is not only used for testing but also here in the source code. This way, modules in `tests` will import from `modalities`, but not the other way around.